servicetalk-grpc-netty/src/main/java/io/servicetalk/grpc/netty/HttpClientBuilderWrapper.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.grpc.netty;
+
+import io.servicetalk.client.api.ConnectionFactory;
+import io.servicetalk.client.api.LoadBalancer;
+import io.servicetalk.client.api.ReservableRequestConcurrencyController;
+import io.servicetalk.client.api.ServiceDiscovererEvent;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.context.api.ContextMap;
+import io.servicetalk.grpc.api.GrpcStatus;
+import io.servicetalk.http.api.DelegatingSingleAddressHttpClientBuilder;
+import io.servicetalk.http.api.FilterableStreamingHttpConnection;
+import io.servicetalk.http.api.FilterableStreamingHttpLoadBalancedConnection;
+import io.servicetalk.http.api.HttpExecutionStrategy;
+import io.servicetalk.http.api.HttpLoadBalancerFactory;
+import io.servicetalk.http.api.SingleAddressHttpClientBuilder;
+import io.servicetalk.http.netty.DefaultHttpLoadBalancerFactory;
+import io.servicetalk.loadbalancer.ErrorClass;
+import io.servicetalk.loadbalancer.RequestTracker;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+import static io.servicetalk.loadbalancer.RequestTracker.REQUEST_TRACKER_KEY;
+
+final class HttpClientBuilderWrapper<U, R> extends DelegatingSingleAddressHttpClientBuilder<U, R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientBuilderWrapper.class);
+
+    private static final Function<GrpcStatus, ErrorClass> PEER_RESPONSE_ERROR_CLASSIFIER = (status) -> {
+        switch (status.code()) {
+            case OK:
+                return null;
```

Comment:
Would a default function be more appropriate?

Reply:
And making it part of the class? We could, although if we want to open this up, either for ourselves or for users to configure more broadly, I think passing in a classification function is much easier to do than having instances methods.